apps/rxjs-race-condition/src/app/topic.service.ts

```diff
@@ -1,10 +1,29 @@
 import { Injectable } from '@angular/core';
-import { map, timer } from 'rxjs';
+import { BehaviorSubject, Observable, map, timer } from 'rxjs';
 
 export type TopicType = 'Politic' | 'Sport' | 'Culture' | 'Nature';
 
 @Injectable({ providedIn: 'root' })
 export class TopicService {
+  topics$ = new BehaviorSubject<{ topics: TopicType[] }>({ topics: [] });
+
+  constructor() {
+    this.init();
+  }
+
   fakeGetHttpTopic = () =>
     timer(1000).pipe(map((): TopicType[] => ['Politic', 'Culture', 'Nature']));
+
+  init() {
+    this.fakeGetHttpTopic().subscribe((topics) => {
+      const data = {
+        topics: topics,
+      };
+      this.topics$.next(data);
+    });
+  }
+
+  getTopics() {
+    return this.topics$;
+  }
```

Comment:
nit: prefer to return the observable and not the subject. 

The subject should not be accessible from the outside. 

Donc create a method, 

just do 
```ts
topics$ = this.topics$$.asObservable()
```

And use $$ or nothing for subject